doc/man3/ASYNC_WAIT_CTX_new.pod

```diff
@@ -103,14 +113,40 @@ code that the job should be resumed by making the wait file descriptor
 "readable". Once resumed the engine should clear the wake signal on the wait
 file descriptor.
 
+Besides file descriptor, "callback" and "status" are provided to implement
+communication between user code and engine. ASYNC_WAIT_CTX_set_callback()
+is used for user code to set a callback, so that engine can fetch the callback
+by calling ASYNC_WAIT_CTX_get_callback() and execute it afterwards. In the
+other direction, engine can talk with user code as well. Engine can report its
+status to user code via ASYNC_WAIT_CTX_set_status(). Unlike file descriptor,
+the "callback" and "status" provides a generic communication channel with is
+independent with system, and since it will not have kernel space and user space
+context switch, it is more efficient.
+
+An example of the usage of callback and status might be the following. User
```

Comment:
I feel the 2 newly added paragraphs need some rewording. I anticipate some English experts could help on this ;-)